worker/rhpv3.go

```diff
@@ -83,31 +89,42 @@ var (
 	errWithdrawalExpired = errors.New("withdrawal request expired")
 )
 
-func isBalanceInsufficient(err error) bool { return isError(err, errBalanceInsufficient) }
-func isBalanceMaxExceeded(err error) bool  { return isError(err, errBalanceMaxExceeded) }
+// IsErrHost indicates whether an error was returned by a host as part of an RPC.
+func IsErrHost(err error) bool {
+	return utils.IsErr(err, errHost)
+}
+
+func isBalanceInsufficient(err error) bool { return utils.IsErr(err, errBalanceInsufficient) }
+func isBalanceMaxExceeded(err error) bool  { return utils.IsErr(err, errBalanceMaxExceeded) }
 func isClosedStream(err error) bool {
-	return isError(err, mux.ErrClosedStream) || isError(err, net.ErrClosed)
+	return utils.IsErr(err, mux.ErrClosedStream) || utils.IsErr(err, net.ErrClosed)
 }
-func isInsufficientFunds(err error) bool  { return isError(err, ErrInsufficientFunds) }
-func isPriceTableExpired(err error) bool  { return isError(err, errPriceTableExpired) }
-func isPriceTableGouging(err error) bool  { return isError(err, errPriceTableGouging) }
-func isPriceTableNotFound(err error) bool { return isError(err, errPriceTableNotFound) }
+func isInsufficientFunds(err error) bool  { return utils.IsErr(err, ErrInsufficientFunds) }
+func isPriceTableExpired(err error) bool  { return utils.IsErr(err, errPriceTableExpired) }
+func isPriceTableGouging(err error) bool  { return utils.IsErr(err, errPriceTableGouging) }
+func isPriceTableNotFound(err error) bool { return utils.IsErr(err, errPriceTableNotFound) }
 func isSectorNotFound(err error) bool {
-	return isError(err, errSectorNotFound) || isError(err, errSectorNotFoundOld)
+	return utils.IsErr(err, errSectorNotFound) || utils.IsErr(err, errSectorNotFoundOld)
 }
-func isWithdrawalsInactive(err error) bool { return isError(err, errWithdrawalsInactive) }
-func isWithdrawalExpired(err error) bool   { return isError(err, errWithdrawalExpired) }
+func isWithdrawalsInactive(err error) bool { return utils.IsErr(err, errWithdrawalsInactive) }
+func isWithdrawalExpired(err error) bool   { return utils.IsErr(err, errWithdrawalExpired) }
 
-func isError(err error, target error) bool {
-	if err == nil {
-		return err == target
+// wrapRPCErr extracts the innermost error, wraps it in either a errHost or
+// errTransport and finally wraps it using the provided fnName.
+func wrapRPCErr(err *error, fnName string) {
+	if *err == nil {
+		return
+	}
+	innerErr := *err
+	for errors.Unwrap(innerErr) != nil {
+		innerErr = errors.Unwrap(innerErr)
 	}
-	// compare error first
-	if errors.Is(err, target) {
-		return true
+	if errors.As(*err, new(*rhpv3.RPCError)) {
+		*err = fmt.Errorf("%w: '%w'", errHost, innerErr)
+	} else {
+		*err = fmt.Errorf("%w: '%w'", errTransport, innerErr)
 	}
-	// then compare the string in case the error was returned by a host
-	return strings.Contains(strings.ToLower(err.Error()), strings.ToLower(target.Error()))
```

Comment:
aren't you concerned this has unpredictable side-effects? You're going from lc-compare to case sensitive compare in `utils.isErr`

Reply:
No, not really. I don't think there was ever a reason for that check to do a lower-case comparison.

Reply:
I'm not sure I caught all of them but check this out:

```
➜  siafoundation ack -Ri 'price table requested is expired|ephemeral account balance was insufficient|ephemeral account maximum balance exceeded|contract has reached the maximum number of revisions|price table not found|could not find the desired sector|sector not found|withdrawal request expired|ephemeral account withdrawals are inactive because the host is not synced' | wc -l
      32
```
```
➜  siafoundation ack -R 'price table requested is expired|ephemeral account balance was insufficient|ephemeral account maximum balance exceeded|contract has reached the maximum number of revisions|price table not found|could not find the desired sector|sector not found|withdrawal request expired|ephemeral account withdrawals are inactive because the host is not synced' | wc -l
      30
```

case sensitive vs case insensitive search showing that `siad` hosts still return

```go
siad/modules/rpc.go
52:	ErrPriceTableNotFound = errors.New("Price table not found")
56:	ErrPriceTableExpired = errors.New("Price table requested is expired")
```